tests/unit/Networking/v2/Models/SubnetTest.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace OpenStack\Test\Subneting\v2\Models;
+
+use GuzzleHttp\Message\Response;
+use OpenStack\Networking\v2\Api;
+use OpenStack\Networking\v2\Models\Subnet;
+use OpenStack\Test\TestCase;
+
+class SubnetTest extends TestCase
+{
+    private $subnet;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->rootFixturesDir = dirname(__DIR__);
+
+        $this->subnet = new Subnet($this->client->reveal(), new Api());
+        $this->subnet->id = 'subnetId';
+    }
+
+    public function test_it_creates()
+    {
+        $opts = [
+            'name' => 'foo',
+            'networkId' => 'networkId',
+            'tenantId' => 'tenantId',
+            'ipVersion' => 4,
+            'cidr' => '192.168.199.0/24',
+            'enableDhcp' => false,
+        ];
+
+        $expectedJson = ['subnet' => [
+            'name' => $opts['name'],
+            'network_id' => $opts['networkId'],
+            'tenant_id' => $opts['tenantId'],
+            'ip_version' => $opts['ipVersion'],
+            'cidr' => $opts['cidr'],
+            'enable_dhcp' => $opts['enableDhcp'],
+        ]];
+
+        $req = $this->setupMockRequest('POST', 'v2.0/subnets', $expectedJson);
+        $this->setupMockResponse($req, 'subnet-post');
+
+        $this->assertInstanceOf(Subnet::class, $this->subnet->create($opts));
+    }
+
+    public function test_it_bulk_creates()
+    {
+        $opts = [
+            [
+                'name' => 'foo',
+                'networkId' => 'networkId',
+                'tenantId' => 'tenantId',
+                'ipVersion' => 4,
+                'cidr' => '192.168.199.0/24',
+            ],
+            [
+                'name' => 'bar',
+                'networkId' => 'networkId',
+                'tenantId' => 'tenantId',
+                'ipVersion' => 4,
+                'cidr' => '10.56.4.0/22',
+            ],
+        ];
+
+        $expectedJson = [
+            'subnets' => [
+                [
+                    'name' => $opts[0]['name'],
+                    'network_id' => $opts[0]['networkId'],
+                    'tenant_id' => $opts[0]['tenantId'],
+                    'ip_version' => $opts[0]['ipVersion'],
+                    'cidr' => $opts[0]['cidr'],
+                ],
+                [
+                    'name' => $opts[1]['name'],
+                    'network_id' => $opts[1]['networkId'],
+                    'tenant_id' => $opts[1]['tenantId'],
+                    'ip_version' => $opts[1]['ipVersion'],
+                    'cidr' => $opts[1]['cidr'],
+                ],
+            ],
+        ];
```

Comment:
I'm thinking maybe we can create a helper method to generate the expected JSON based off the input options. For example:

```php
$expectedJson = $this->expectedJson($opts, [
   'networkId' => 'network_id',
   'ipVersion'  => 'ip_version'
]);
```

As you can see, aliases are passed in as the second arg. Or, even better, why not pass in the operation data? We'd need to add a new class property:

```php
public function __construct()
{
   // existing stuff
   $this->api = new Api();
   $this->subnet = new Subnet($this->client->reveal(), $this->api);
   $this->subnet->id = '{subnetId}';
}

public function test_it_bulk_creates()
{
   $opts = [];

   $expectedJson = $this->expectedJson($opts, $this->api->postSubnets()['params']);
}
```
I can add this in the future, what do you think? Worth it or not?